supervisord.conf

```diff
@@ -10,7 +10,10 @@ stdout_logfile_maxbytes = 50000000
 stdout_logfile_backups = 3
 stderr_logfile = /dev/stderr
 stderr_logfile_maxbytes = 0
-stopsignal = INT
+stopsignal = STOP
+# bosh will probably kill everything after 10s - stop a second early in the hopes that the logs of our
+# last handled request make it off the machine
+stopwaitsecs = 9
```

Comment:
Should we kill nginx before uwsgi to make sure no new connections are created before the kill? Or does bosh handle this for us?

Reply:
CF (I'm not sure which component) won't give us any new connections after it's given us `SIGTERM`. I'm not sure it matters about order when we get to the 9s limit, because the 9s limit is an ugly `SIGKILL` - there's no grace at this point, the best we can do is give the logs a chance to get off the machine.

Reply:
> CF (I'm not sure which component) won't give us any new connections after it's given us SIGTERM.

Is there documentation for that? Just being paranoid.

Reply:
Well, if it didn't behave like that there wouldn't be a hope in hell of any cf app ever shutting down gracefully.

Reply:
Yeah... that's what I'm paranoid about...

Reply:
...and there wouldn't be any point in sending us a 10s warning signal, because there's nothing we could do about it and the web would be chock full of "cf can't cleanly shutdown apps" complaints.

Reply:
But it's not documented?

Reply:
The [app lifecycle page](https://docs.cloudfoundry.org/devguide/deploy-apps/app-lifecycle.html) seems to suggest that it's our responsibility to stop accepting new requests :/

> Apps must finish their in-flight jobs within ten seconds of receiving the SIGTERM before CF terminates the app with a SIGKILL. For instance, a web app must finish processing existing requests and stop accepting new requests.

Reply:
I see, I misunderstood the meaning of stopwaitsecs.